modules/blockexplorer/addblock.go

```diff
@@ -172,7 +173,17 @@ func (be *BlockExplorer) addBlockDB(b types.Block) error {
 	}
 	defer tx.Rollback()
 
-	// Construct the struct that will be inside the database
+	// Check if the block is already there before adding it
+	blkBytes, err := be.db.GetFromBucket("Blocks", encoding.Marshal(b.ID()))
+	if err != nil && err != persist.ErrNilEntry {
+		return err
+	}
```

Comment:
whoops, I didn't realize you already had a `GetFromBucket` function. It kinda overlaps with `getObject`.
fine for now, but maybe later we could add an `exists` helper function that checks for an entry.